src/typegoose.ts

```diff
@@ -44,17 +44,7 @@ export class Typegoose {
   ) {
     const name = this.constructor.name;
 
-    // get schema of current model
-    let sch = this.buildSchema<T>(t, name, schemaOptions);
-    // get parents class name
-    let parentCtor = Object.getPrototypeOf(this.constructor.prototype).constructor;
-    // iterate trough all parents
-    while (parentCtor && parentCtor.name !== 'Typegoose' && parentCtor.name !== 'Object') {
-      // extend schema
-      sch = this.buildSchema<T>(t, parentCtor.name, schemaOptions, sch);
-      // next parent
-      parentCtor = Object.getPrototypeOf(parentCtor.prototype).constructor;
-    }
+    const sch = this.buildPublicSchema<T>(t, { existingMongoose, schemaOptions });
```

Comment:
i would recommend renaming `sch` to `schema` (i know that it was already named so)

Reply:
doesn't seem to be much of importance, there is an import at the top level called schema, so that's the reason for the original use of sch. but it's clear enough if to be honest.

Reply:
didnt see the import, ok, just thought it would be a good idea
(you can mark all 3 now as resolved)